File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/InspectionPlan.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// InspectionPlan Declaration
//==============================================================
type InspectionPlan struct {
    gorm.Model
     PlanNumber                                    string
    Revision                                    string
    ItemId         *uint
    Item           *Item `gorm:"foreignKey:ItemId"`
     Characteristics           []InspectionCharacteristic `gorm:"foreignKey:CharacteristicsFromInspectionPlanId"`
    SamplingPlan                      SamplingPlanType
    Status                      QualityPlanStatus

// parent associations as their child

}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/InspectionPlan.py
from django.db import models
from manufacturingOnDjango.models.SamplingPlanType import SamplingPlanType
from manufacturingOnDjango.models.QualityPlanStatus import QualityPlanStatus

#======================================================================
# 
# Encapsulates data for model InspectionPlan
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class InspectionPlan Declaration
#======================================================================
class InspectionPlan (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	planNumber = models.CharField(max_length=200, null=True)
	revision = models.CharField(max_length=200, null=True)
	item = models.ForeignKey('Item', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	characteristics = models.ManyToManyField('InspectionCharacteristic',  blank=True, related_name='+')
	samplingPlan = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in SamplingPlanType])
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in QualityPlanStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.planNumber
		str = str + self.revision
		str = str + self.samplingPlan
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "InspectionPlan";
    
	def objectType(self):
		return "InspectionPlan";
